### DISCUSSION TITLE
Lesson 17: Never Use A Env File

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#1707] [Q&A] [answered]
GM Fam
we encrypt our private key using this command :
```
cast wallet import NAME_OF_FILE --interactive 
```
and we use it in our `forge` commands using `--account` switch.
but i saw patrick used `--sender` too, and i don't have any idea what is it and why he used it.

**Comment by MostPow3rful** [comment]
and another question is what was the output of `cast wallet` command? i mean what is that address and where should i use ?

**Reply by EngrPips** [reply, marked as answer]
Hello @MostPow3rful, The `--sender` flag is used to specify who the sender of the current cast call would be, and the address you get when you do `cast wallet` is the address that you would produce whenever you want to use the private key you encrypted. When you produce that address to forge, it will know which private is linked to that address and help decode it after successfully providing the password.

**Reply by EngrPips** [reply]
If you think anything is still not clear enough please let me know so I can maybe explain in a better way.

**Reply by MostPow3rful** [reply]
GM @EngrPips . thanks for your explain but i still don't understand this.
can you explain it with an example or senario ?
if we don't set `--sender`, will throw an error or something miss ?
and i checked the return of `cast wallet` again. it returns the public key of private key. am i right ?

**Reply by EngrPips** [reply]
It returns a public key associated with the private key you encrypted, which would always be passed in after the `--sender` flag so it would know the private key you want to use for the transaction you are about to send through the terminal. Do you think this makes sense? You should try running a `cast send` command without the `--sender` flag and see what will happen.

**Reply by MostPow3rful** [reply]
i used many time foundry toolkit and tested the `cast send` and `cast call`. since i learned this `cast wallet` option and et., i got confused.
but i will try some command. hope to understand this.
thanks a lot for your answer

**Reply by EngrPips** [reply]
Please feel free to reach out here if anything needs to be clarified when you try. I will be glad to explain more if you need me to.

**Reply by Robschuh** [reply]
Hi there @MostPow3rful  , here is a short example about this topic.
The output generated after running the command you mentioned above (cast wallet) is the address of the account you are encrypted its PRIVATE KEY. Example:
![address](https://github.com/Cyfrin/foundry-full-course-f23/assets/142880927/7f37318b-b18d-4802-ae26-d72bb259506b)
This Address: 0xa0ee7a142d267c1f36714e4a8f75612f20a79720 is only the address of that account NOT the private key or a public key.

**Reply by MostPow3rful** [reply]
thanks a lot sir

**Comment by SirDivider** [comment]
thx for it!

**Comment by EngrPips** [comment]
Interesting.
